seisbench/data/base.py

```diff
@@ -2677,3 +2706,53 @@ def flush_hdf5(self):
         for bucket in buckets:
             self._write_bucket(self._cache[bucket])
             del self._cache[bucket]
+
+
+class EventParameters(TypedDict):
+    index: NotRequired[int]
+    split: Literal["train", "dev", "test"]
+
+    source_id: str
```

Comment:
Can you label all of these parameters as `NotRequired`. The data format is supposed to be very permissive.